scripts/dts/extract/default.py

```diff
@@ -64,6 +64,12 @@ def extract(self, node_path, prop, prop_type, def_label):
 
         insert_defs(node_path, prop_def, prop_alias)
 
+        if prop_type == 'string':
+            new_prop = {}
+            prop_val = label + '_' + str_to_label(prop_def[label]).strip('"')
+            new_prop[prop_val] = '1'
+            insert_defs(node_path, new_prop, {})
```

Comment:
Could shorten this down to

```python
prop_val = label + '_' + str_to_label(prop_def[label]).strip('"')
insert_defs(node_path, {prop_val: '1'}, {})
```

or the like and get rid of `new_prop`. `prop_val` seems like a weird variable name for a macro name too, but maybe it's consistent with some other thing.

I want to replace `insert_defs()` with a function that just inserts a single name/value at some point. It's awkward that you have to call it with dictionaries.

Reply:
Wonder if we could start a better tradition in the DTS code and add a comment too, mentioning what kind of stuff this generates, and why it's handy.

Reply:
@ulfalizer I can add comments describing the change, I just wanted to get feedback about the proposed solution.

Reply:
FWIW, at least GCC is clever enough to optimize away `strcmp()`s with constant arguments:

```c
#include <string.h>

#define FOO "hello"

int f(void)
{
    return strcmp(FOO, "hello") ? 0x123 : 0x456;
}
```

gives

```
   0:	b8 56 04 00 00       	mov    eax,0x456
   5:	c3                   	ret    
```

Could use `__builtin_strcmp()` to make sure GCC's builtin version is used.

If we don't want to rely on that, and don't want to have case-by-case code for string properties, then something like this PR is probably the only option.

Reply:
The point was to use it in the preprocessor, can we use strcmp there?

Reply:
> The point was to use it in the preprocessor, can we use strcmp there?

Nah, won't be as general.

Could have a zero-overhead `IS_ENABLED()`-like though.

Reply:
> > The point was to use it in the preprocessor, can we use strcmp there?
> 
> Nah, won't be as general.
> 
> Could have a zero-overhead `IS_ENABLED()`-like though.

Do you mean we can use somehow strings in preprocessor?

Reply:
@finikorg 
I mean that if you just need to conditionally run some code depending on the value of a DT string property, then it could be implemented with zero overhead by relying on the compiler to optimize away `strcmp()`s with constant arguments, similar to how `IS_ENABLED()` works for Kconfig symbols.

```c
if (DT_STREQ(MAXIMUM_SPEED, "full-speed")) {
        ...
} else if (DT_STREQ(MAXIMUM_SPEED, "yadayada")) {
        ...
```

`DT_STREQ` could use `__builtin_strcmp()` or similar, if we assume GCC/Clang.

I don't know how reliable that is with other compilers, and it's less general than your solution. Just another option.

Reply:
@ulfalizer That strings should be evaluated only once so we can leave strcmp as is since it is much more readable.

Reply:
Should just be careful so that the compiler doesn't turn it into an actual call in that case, for whatever reason ("hmm, perhaps this isn't guaranteed to be the standard `strcmp()`, due to compilation setting `-foo`, so I can't safely optimize it").

The `__builtin_` variants are nice in that they're guaranteed to avoid stuff like that.

Reply:
That is probably needs to be done in `strings` or wherever `strcmp` is implemented.

Reply:
> Could shorten this down to
> 
> ```python
> prop_val = label + '_' + str_to_label(prop_def[label]).strip('"')
> insert_defs(node_path, {prop_val: '1'}, {})
> ```
> 
> or the like and get rid of `new_prop`.
> 
> `prop_val` seems like a weird variable name for a macro name too, but maybe it's consistent with some other thing. Could it be called `name` or something instead?
> 
> I want to replace `insert_defs()` with a function that just inserts a single name/value at some point. It's awkward that you have to call it with dictionaries.

simplified this way